### DISCUSSION TITLE
Attaching previously provisioned storage

### DISCUSSION THREAD

**Discussion by akib1689** [strimzi/strimzi-kafka-operator#10788] [Q&A] [answered]
We are hosting a simple kafka cluster with 3 broker pod and 3 zookeeper pod into our company. Currently the storage is dynamically provisioned during creation of the cluster. however I want this business logic:

> [!NOTE]
> I want to provision the azure blob storage (preferably with `openTofu` or `pulumi`).
> Then I want the strimzi operator to use those provisioned disk as the broker and zookeeper storage. Every tutorial and documentation dynamically provision those storage.

Has Anyone worked with this specific use case? I'm at a loss here.

**Comment by scholzj** [comment, marked as answer]
I do not really understand what exactly are you trying to do. But if you pre-create the PVCs for Strimzi with their expected names, it will just reuse them. So that is what you can do. Just keep in mind that you will likely not be able to reuse some existing volumes from another cluster because of paths, configurations etc.

**Reply by scholzj** [reply]
BTW: Azure Blob storage as IIRC object storage. So it is absolutely unsuitable for use with Kafka. If you want to use object storage for part of your Kafka data, you should consider using Kafka tiered storage. But for the regular disks, you need block storage, and Azure Managed Disks is what you should be using.

**Comment by akib1689** [comment]
Thanks a lot I'll give it a try and see if it works
